sources/packages/backend/libs/integrations/src/services/msfaa-number/msfaa-number.service.ts

```diff
@@ -191,22 +196,56 @@ export class MSFAANumberService extends RecordDataModelService<MSFAANumber> {
     cancelledDate: Date,
     newIssuingProvince: string,
   ): Promise<UpdateResult> {
-    if (!cancelledDate || !newIssuingProvince) {
-      throw new Error(
-        "Not all required fields to update a received MSFAA record were provided.",
+    return this.dataSource.transaction(async () => {
+      // Keeping both MSFAA Cancellation update and saving of MSFAA Notification message as a part of the same transaction.
+      if (!cancelledDate || !newIssuingProvince) {
+        throw new Error(
+          "Not all required fields to update a received MSFAA record were provided.",
+        );
+      }
+      // Update the MSFAA record that needs to be cancelled.
+      const updateResult = await this.repo.update(
```

Comment:
Please review how other transactions are executed, usually, they are declared as below.
`await this.dataSource.transaction(async (transactionalEntityManager)`
If you was able to check that the current code is also working and the codes are part of the same transaction, can you please provide further information?

Reply:
You are right. I missed to provide the ready-to-use `transactionalEntityManager` as an argument to this `transaction` callback. Corrected it. Thanks 👍

Reply:
This must use the transaction manager. Please see other similar codes along the source code for reference.
![image](https://github.com/bcgov/SIMS/assets/61259237/8e293fa7-6481-469a-a70a-d8f88624d822)